src: Add information about ICPC (#2)
obiwac/uclouvain-icpc: public/md/index.md

- You (sometimes) receive **balloons** when you successfully complete a problem.
- Karaoke. Need I say more?
- It's a perfect addition to your **Tinder** profile!

> ## What is ICPC

Reviewer: ```suggestion
## What is ICPC?

```

> ICPC stands for International Collegiate Programming Contest. It is an algorithmic programming contest for college students. Teams of three, representing their university, work to solve the most real-world problems, fostering collaboration, creativity, innovation, and the ability to perform under pressure.

Reviewer: ```suggestion
ICPC (International Collegiate Programming Contest) is an algorithmics programming contest for university students.
Teams of three - representing their university - work to solve the most problems, fostering collaboration, creativity, innovation, and the ability to perform under pressure.
```

"college" c'est très USA-centrique, stuv tu peux mettre des trucs en bold pour un peu de couleur,

> ## How it works ? 

Reviewer: ```suggestion
## How does it work?

```

> There is multiple levels of competition. Here in Belgium there is 3 levels:
> - BAPC
> - NWERC
> - Worlds Finals

Reviewer: ```suggestion
Here in Belgium there are 3 levels to go through before reaching the world finals:

- preBAPC
- BAPC
- NWERC
- World finals
```

Peut-être aussi parler du KARWa qqpart là dedans comme option pour les francophones?

> ### BAPC
> BAPC stands for Benelux Algorithm Programming Contest

Reviewer: ```suggestion
### preBAPC & BAPC

BAPC (Benelux Algorithm Programming Contest).
```

Ce serait cool d'avoir un peu plus à dire là dessus mais j'imagine que you're on it :)

> ### NWERC
> NWERC stands for Northwestern Europe Regional Contest

Reviewer: ```suggestion
### NWERC

NWERC (Northwestern Europe Regional Contest).
```

> ### Worlds Finals

Reviewer: ```suggestion
### World finals
```

- You (sometimes) receive **balloons** when you successfully complete a problem.
- Karaoke. Need I say more?
- It's a perfect addition to your **Tinder** profile!

## What is ICPC?

ICPC (International Collegiate Programming Contest) is an algorithmics programming contest for university students.
Teams of three - representing their university - work to solve the most problems, fostering collaboration, creativity, innovation, and the ability to perform under pressure.

## How does it work?

Here in Belgium there are 3 levels to go through before reaching the world finals:

> - pre-BAPC

Reviewer: ```suggestion
- Pre-BAPC
```

- BAPC
- NWERC
- World finals

Each contest has on average 12 problems from all difficulties.

> Eeach problem has a Time and memory constrain, to solve this problem your solution must fit the constrains.

Reviewer: ```suggestion
Each contest has an average of 12 problems ranging from easy (-ish) to hard, though the NWERC is generally harder than the BAPC which is generally harder than the pre-BAPC.
Each problem has a time limit and a memory constraint, which your solution must execute within.
```

### pre-BAPC & BAPC

> Pre-BAPC is a competition which occurs inside of the [University Catholique de  Louvain](https://en.wikipedia.org/wiki/Universit%C3%A9_catholique_de_Louvain). The best teams will go to the BAPC.

Reviewer: ```suggestion
### Pre-BAPC & BAPC

The Pre-BAPC is a qualifying round which takes place in each university participating in the BAPC (in our case, [University Catholique de  Louvain](https://en.wikipedia.org/wiki/Universit%C3%A9_catholique_de_Louvain)).
The best teams will go on to participate in the BAPC.
```

> BAPC (Benelux Algorithm Programming Contest).

Reviewer: ya encore à faire ici non?

### NWERC

> NWERC (Northwestern Europe Regional Contest) is one of the most important contest, University from Northwestern Europe will compote against each other. Generally top 3 teams will go to the Worlds finals.

Reviewer: ```suggestion
NWERC (Northwestern Europe Regional Contest) is the contest where universities across Northwestern Europe compete for a spot in the ICPC world finals.
Generally it's the best 3 teams who go to the world finals.
Doing well in the BAPC isn't necessarily a prerequisite for being sent to the NWERC; that's a decision the university makes.
```

> ### World finals
> ...
> ## Training
> Eeach week, our lovely coach Henri prepare a weekly contest on [Kattis](https://open.kattis.com/). Every thursday we discuss about the problems and solutions.
> ## How to participate
> You can join us via [Discord](https://discord.gg/HYVNGM9yGw)
> 

Reviewer: ```suggestion
### World finals

...

## Training

Each week, our beloved coach Henri prepares a weekly contest on [Kattis](https://open.kattis.com/).
Every Thursday at lunchtime we discuss the problems and solutions together.

## How to participate

You can join us on [Discord](https://discord.gg/HYVNGM9yGw)!
```

pour les liens vers discord &co, je pense que je vais faire ça côté rust

Here in Belgium there are 3 levels to go through before reaching the world finals:

- Pre-BAPC
- [BAPC](https://en.wikipedia.org/wiki/BAPC)
- [NWERC](https://2023.nwerc.eu/)

> - World finals

Reviewer: link for this one too?

Each contest has an average of 12 problems ranging from easy (-ish) to hard, though the NWERC is generally harder than the BAPC which is generally harder than the pre-BAPC.
Each problem has a time limit and a memory constraint, which your solution must execute within.

> ### When ?
> - Pre-BAPC & BAPC generally occurs in october
> - NWERC occurs in late november or early december.
> 
> 

Reviewer: ```suggestion
### When ?

- The Pre-BAPC & BAPC generally occur in October
- The NWERC occurs in late November or early December.

```

peut-être mieux de mettre ça dans les grosses sections `# BAPC` et `# NWERC`? jsp

### World finals

> World finals take place all around the world, it is the most presigious contest. Contest rules are the same as NWERC & BAPC.
> Problem set tends to be more difficult than NWERC.
> 

Reviewer: ```suggestion
World finals can take place anywhere around the world, and is the most prestigious of the contests.
The contest rules are the same as the NWERC and the BAPC, but the problem set tends to be harder than the NWERC.
```

## Training

Each week, our beloved coach Henri prepares a weekly contest on [Kattis](https://open.kattis.com/).
Every Thursday at lunchtime we discuss the problems and solutions together.

> There is a lot of ressources to learn competitive programming, some website handle online contest to train : 
> - [Codeforces](https://codeforces.com/)

Reviewer: ```suggestion
There are a lot of resources to learn competitive programming.
Some website even host online contests to train on.

Here are a few of them:
 
- [Codeforces](https://codeforces.com/)
```

- [Atcoder](https://atcoder.jp/)
- [Kattis](https://open.kattis.com/)
- [Codechef](https://www.codechef.com/)
- [CSES](https://cses.fi/)

> There is also books where you can lean the basis
> - [CSES](https://cses.fi/book/index.php)

Reviewer: ```suggestion
There are also books from which you can learn the basics:

- [CSES](https://cses.fi/book/index.php)
```
